crates/core/component/ibc/src/component/rpc/connection_query.rs

```diff
@@ -58,16 +59,19 @@ impl<HI: HostInterface + Send + Sync + 'static> ConnectionQuery for IbcQuery<HI>
         let conn =
             conn.map_err(|e| tonic::Status::aborted(format!("couldn't decode connection: {e}")))?;
 
-        let height = Height {
-            revision_number: 0,
-            revision_height: snapshot.version(),
-        };
-
-        let res = QueryConnectionResponse {
-            connection: conn,
-            proof: proof.encode_to_vec(),
-            proof_height: Some(height),
-        };
+        let res =
+            QueryConnectionResponse {
+                connection: conn,
+                proof: proof.encode_to_vec(),
+                proof_height: Some(ibc_proto::ibc::core::client::v1::Height {
+                    revision_height: snapshot.get_block_height().await.map_err(|e| {
+                        tonic::Status::aborted(format!("couldn't decode height: {e}"))
+                    })? + 1,
+                    revision_number: HI::get_revision_number(&snapshot).await.map_err(|e| {
+                        tonic::Status::aborted(format!("couldn't decode height: {e}"))
+                    })?,
+                }),
+            };
```

Comment:
Regarding the `get_block_height + 1` logic throughout, that unblocked so much of these state machine tests, we should be documenting those sharp edges more comprehensively. Not blocking merge, but IMO, explaining the difference in height queries in the ibc module docs makes the most sense: it'll be immediately obvious to any developers looking at that code that the various notions of height are important and distinct.